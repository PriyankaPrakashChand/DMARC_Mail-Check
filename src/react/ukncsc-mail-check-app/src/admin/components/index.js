export { default as AddEntitiesModal } from './AddEntitiesModal';
export { default as DomainAdmin } from './DomainAdmin';
export { default as EntitiesSearch } from './EntitiesSearch';
export { default as EntityAccessGroup } from './EntityAccessGroup';
export { default as EntityDetails } from './EntityDetails';
export { default as EntityDisplay } from './EntityDisplay';
export {
  default as EntitySearchableDropdown,
} from './EntitySearchableDropdown';
export { default as GroupAdmin } from './GroupAdmin';
export { default as UserAdmin } from './UserAdmin';
export { default as withAddRemoveModals } from './withAddRemoveModals';
