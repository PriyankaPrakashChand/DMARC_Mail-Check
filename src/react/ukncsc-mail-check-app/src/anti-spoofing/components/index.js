export { default as DomainsStatusDisplay } from './DomainsStatusDisplay';
export { default as Pagination } from './Pagination';
export { default as RatingDisplay } from './RatingDisplay';
