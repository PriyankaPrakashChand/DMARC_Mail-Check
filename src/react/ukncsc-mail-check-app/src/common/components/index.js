export { default as Breadcrumb } from './Breadcrumb';
export { default as BreadcrumbItem } from './BreadcrumbItem';
export { default as ConfirmationModal } from './ConfirmationModal';
export { default as DetailedTab } from './DetailedTab';
export { default as DetailedTabList } from './DetailedTabList';
export { default as IeFriendlyVictoryChart } from './IeFriendlyVictoryChart';
export { default as MailCheckMessage } from './MailCheckMessage';
export { default as ScrollToTop } from './ScrollToTop';
export { default as ShowMoreDropdown } from './ShowMoreDropdown';
export { default as TermsAndConditions } from './TermsAndConditions';
