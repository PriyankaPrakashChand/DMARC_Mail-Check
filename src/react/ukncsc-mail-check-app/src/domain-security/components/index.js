export { default as AggregateReportChart } from './AggregateReportChart';
export {
  default as AggregateReportChartLegend,
} from './AggregateReportChartLegend';
export {
  default as AggregateReportDateSelector,
} from './AggregateReportDateSelector';
export { default as AggregateReportSummary } from './AggregateReportSummary';
export {
  default as DomainSecurityDetailsMessages,
} from './DomainSecurityDetailsMessages';
export { default as DomainSecurityDetailsMx } from './DomainSecurityDetailsMx';
export {
  default as DomainSecurityDetailsTxt,
} from './DomainSecurityDetailsTxt';
export { default as DomainSecurityPill } from './DomainSecurityPill';
export { default as DomainSecurityRecord } from './DomainSecurityRecord';
export {
  default as DomainSecurityRecordExplanation,
} from './DomainSecurityRecordExplanation';
export { default as DomainSecuritySummary } from './DomainSecuritySummary';
export {
  default as DomainSecuritySummaryMessages,
} from './DomainSecuritySummaryMessages';
export { default as DomainSecuritySummaryMx } from './DomainSecuritySummaryMx';
export {
  default as DomainSecuritySummarySubdomains,
} from './DomainSecuritySummarySubdomains';
export {
  default as DomainSecuritySummaryTxt,
} from './DomainSecuritySummaryTxt';
export { default as DomainSecurityTitle } from './DomainSecurityTitle';
