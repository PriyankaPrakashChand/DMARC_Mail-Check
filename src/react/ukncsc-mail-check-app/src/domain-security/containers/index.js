export {
  default as AggregateReportSummaryContainer,
} from './AggregateReportSummaryContainer';
export {
  default as DomainSecurityAggregateExport,
} from './DomainSecurityAggregateExport';
export {
  default as DomainSecurityDetailsCertificates,
} from './DomainSecurityDetailsCertificates';
export {
  default as DomainSecurityDetailsDkimContainer,
} from './DomainSecurityDetailsDkimContainer';
export {
  default as DomainSecurityDetailsDmarc,
} from './DomainSecurityDetailsDmarc';
export {
  default as DomainSecurityDetailsSpf,
} from './DomainSecurityDetailsSpf';
export {
  default as DomainSecurityDetailsTls,
} from './DomainSecurityDetailsTls';
export {
  default as DomainSecuritySummaryCertificatesContainer,
} from './DomainSecuritySummaryCertificatesContainer';
export {
  default as DomainSecuritySummaryContainer,
} from './DomainSecuritySummaryContainer';
export {
  default as DomainSecuritySummaryDkimContainer,
} from './DomainSecuritySummaryDkimContainer';
export {
  default as DomainSecuritySummaryDmarcContainer,
} from './DomainSecuritySummaryDmarcContainer';
export {
  default as DomainSecuritySummarySpfContainer,
} from './DomainSecuritySummarySpfContainer';
export {
  default as DomainSecuritySummarySubdomainsContainer,
} from './DomainSecuritySummarySubdomainsContainer';
export {
  default as DomainSecuritySummaryTlsContainer,
} from './DomainSecuritySummaryTlsContainer';
export { default as DomainSecurityTlsAdvice } from './DomainSecurityTlsAdvice';
