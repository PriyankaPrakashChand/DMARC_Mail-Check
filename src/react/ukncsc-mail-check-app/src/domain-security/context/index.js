export {
  default as DomainSecurityLocationContext,
} from './DomainSecurityLocationContext';
