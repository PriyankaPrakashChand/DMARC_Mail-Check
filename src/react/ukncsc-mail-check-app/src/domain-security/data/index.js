export * from './domain-security-copy';
export * from './aggregate-report-info-static-data';
export * from './aggregate-report-export-fields';
