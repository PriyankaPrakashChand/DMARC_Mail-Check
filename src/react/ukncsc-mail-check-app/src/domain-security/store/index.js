import { combineReducers } from 'redux';

import aggregateData from './domain-security-aggregate-data';
import certificates from './domain-security-certificates';
import dmarc from './domain-security-dmarc';
import dkim from './domain-security-dkim';
import domain from './domain-security-domain';
import spf from './domain-security-spf';
import subdomains from './domain-security-subdomains';
import tls from './domain-security-tls';

export * from './domain-security-aggregate-data';
export * from './domain-security-certificates';
export * from './domain-security-dkim';
export * from './domain-security-dmarc';
export * from './domain-security-domain';
export * from './domain-security-spf';
export * from './domain-security-subdomains';
export * from './domain-security-tls';

export default combineReducers({
  aggregateData,
  certificates,
  dkim,
  dmarc,
  domain,
  spf,
  subdomains,
  tls,
});
