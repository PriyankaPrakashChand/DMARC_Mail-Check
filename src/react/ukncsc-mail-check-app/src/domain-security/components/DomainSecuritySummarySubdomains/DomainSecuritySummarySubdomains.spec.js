import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';

import { DomainSecuritySummarySubdomains } from 'domain-security/components';

const fetchDomainSecuritySubdomains = jest.fn();
const getDomainSecuritySubdomains = jest.fn();

describe('DomainSecuritySubdomains', () => {
  let container;

  afterEach(() => jest.resetAllMocks());

  describe('when there is empty subdomains', () => {
    beforeEach(() => {
      getDomainSecuritySubdomains.mockImplementation(() => ({
        loading: false,
        noMoreResults: true,
        noSubdomains: false,
        subdomains: [],
      }));

      ({ container } = render(
        <MemoryRouter>
          <DomainSecuritySummarySubdomains
            domainName="ncsc.gov.uk"
            fetchDomainSecuritySubdomains={fetchDomainSecuritySubdomains}
            getDomainSecuritySubdomains={getDomainSecuritySubdomains}
          />
        </MemoryRouter>
      ));
    });

    test('it should not display a show more link', () =>
      expect(container).not.toHaveTextContent('Show more'));

    test('it should not display anything in the list', () =>
      expect(
        container.getElementsByClassName('eight wide column')[0].firstChild
      ).toBeNull());
  });

  describe('when there are no subdomains', () => {
    beforeEach(() => {
      getDomainSecuritySubdomains.mockImplementation(() => ({
        loading: false,
        noMoreResults: true,
        noSubdomains: true,
        subdomains: [],
      }));

      ({ container } = render(
        <MemoryRouter>
          <DomainSecuritySummarySubdomains
            domainName="ncsc.gov.uk"
            fetchDomainSecuritySubdomains={fetchDomainSecuritySubdomains}
            getDomainSecuritySubdomains={getDomainSecuritySubdomains}
          />
        </MemoryRouter>
      ));
    });

    test('it should not display a show more link', () =>
      expect(container).not.toHaveTextContent('Show more'));

    test('it should display an informative message', () =>
      expect(container).not.toBeNull());

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when there are subdomains with more results', () => {
    beforeEach(() => {
      getDomainSecuritySubdomains.mockImplementation(() => ({
        loading: false,
        noMoreResults: false,
        subdomains: [{ id: 123, domainName: 'ncsc.gov.uk' }],
      }));

      ({ container } = render(
        <MemoryRouter>
          <DomainSecuritySummarySubdomains
            domainName="ncsc.gov.uk"
            fetchDomainSecuritySubdomains={fetchDomainSecuritySubdomains}
            getDomainSecuritySubdomains={getDomainSecuritySubdomains}
          />
        </MemoryRouter>
      ));
    });

    test('it should display a show more link', () =>
      expect(container).toHaveTextContent('Show more'));

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when there are subdomains with no more results', () => {
    beforeEach(() => {
      getDomainSecuritySubdomains.mockImplementation(() => ({
        loading: false,
        noMoreResults: true,
        subdomains: [{ id: 123, domainName: 'ncsc.gov.uk' }],
      }));

      ({ container } = render(
        <MemoryRouter>
          <DomainSecuritySummarySubdomains
            domainName="ncsc.gov.uk"
            fetchDomainSecuritySubdomains={fetchDomainSecuritySubdomains}
            getDomainSecuritySubdomains={getDomainSecuritySubdomains}
          />
        </MemoryRouter>
      ));
    });

    test('it should not display a show more link', () =>
      expect(container).not.toHaveTextContent('Show more'));

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
