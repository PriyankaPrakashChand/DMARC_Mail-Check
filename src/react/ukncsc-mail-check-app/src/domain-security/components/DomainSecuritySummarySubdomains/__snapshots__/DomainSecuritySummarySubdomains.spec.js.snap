// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecuritySubdomains when there are no subdomains it should match the snapshot 1`] = `
<div>
  <div
    class="ui stackable grid"
  >
    <div
      class="row"
    >
      <div
        class="four wide column"
      >
        <h2
          class="ui header DomainSecurityTitle--header"
        >
          Subdomains
        </h2>
        <div
          class="DomainSecurityTitle--icon"
        />
      </div>
      <div
        class="eight wide column"
      >
        <div
          style="margin: 0.2em 0em; max-width: 600px;"
          class="ui info message MailCheckMessage--container"
        >
          We're currently not checking any subdomains for ncsc.gov.uk.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DomainSecuritySubdomains when there are subdomains with more results it should match the snapshot 1`] = `
<div>
  <div
    class="ui stackable grid"
  >
    <div
      class="row"
    >
      <div
        class="four wide column"
      >
        <h2
          class="ui header DomainSecurityTitle--header"
        >
          Subdomains
        </h2>
        <div
          class="DomainSecurityTitle--icon"
        />
      </div>
      <div
        class="eight wide column"
      >
        <h4
          style="display: inline-block; margin-top: 0px;"
          class="ui header"
        >
          ncsc.gov.uk
        </h4>
        <p>
          <a
            href="/my-domains/123"
          >
            View information
          </a>
        </p>
        <div
          class="ui divider"
        />
        <p>
          <a
            href=""
          >
            Show more
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`DomainSecuritySubdomains when there are subdomains with no more results it should match the snapshot 1`] = `
<div>
  <div
    class="ui stackable grid"
  >
    <div
      class="row"
    >
      <div
        class="four wide column"
      >
        <h2
          class="ui header DomainSecurityTitle--header"
        >
          Subdomains
        </h2>
        <div
          class="DomainSecurityTitle--icon"
        />
      </div>
      <div
        class="eight wide column"
      >
        <h4
          style="display: inline-block; margin-top: 0px;"
          class="ui header"
        >
          ncsc.gov.uk
        </h4>
        <p>
          <a
            href="/my-domains/123"
          >
            View information
          </a>
        </p>
        <div
          class="ui divider"
        />
      </div>
    </div>
  </div>
</div>
`;
